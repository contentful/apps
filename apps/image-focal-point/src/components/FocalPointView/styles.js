import { css } from 'emotion';
import tokens from '@contentful/forma-36-tokens';

export const styles = {
  container: css({
    display: 'flex',
    alignItems: 'center'
  }),
  input: css({
    maxWidth: '280px'
  }),
  button: css({
    marginLeft: tokens.spacingXs,
    marginRight: tokens.spacingXs
  })
};
