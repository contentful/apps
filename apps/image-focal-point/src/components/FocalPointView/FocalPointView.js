import React from 'react';
import PropTypes from 'prop-types';
import { Button, TextInput, TextLink } from '@contentful/forma-36-react-components';

import { styles } from './styles';

const FocalPointView = ({ focalPoint, showFocalPointDialog, resetFocalPoint }) => {
  const value = focalPoint ? `x: ${focalPoint.x}px / y: ${focalPoint.y}px` : 'Focal point not set';

  return (
    <div className={styles.container}>
      <TextInput
        className={styles.input}
        width="large"
        type="text"
        id="focal-point"
        testId="focal-point"
        value={value}
        disabled
      />
      <Button className={styles.button} buttonType="muted" onClick={showFocalPointDialog}>
        Set focal point
      </Button>
      <TextLink linkType="primary" onClick={() => resetFocalPoint()}>
        Reset focal point
      </TextLink>
    </div>
  );
};

FocalPointView.propTypes = {
  focalPoint: PropTypes.shape({
    x: PropTypes.number.isRequired,
    y: PropTypes.number.isRequired
  }),
  showFocalPointDialog: PropTypes.func.isRequired,
  resetFocalPoint: PropTypes.func.isRequired
};

FocalPointView.defaultProps = {
  focalPoint: {
    x: 0,
    y: 0
  }
};

export { FocalPointView };
