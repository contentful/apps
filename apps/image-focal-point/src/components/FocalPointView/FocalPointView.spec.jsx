import React from 'react';
import { render } from '@testing-library/react';
import { vi } from 'vitest';

import { FocalPointView } from './FocalPointView';

const props = {
  focalPoint: {
    x: 10,
    y: 30,
  },
  showFocalPointDialog: vi.fn(),
  resetFocalPoint: vi.fn(),
};

describe('FocalPointView', () => {
  it('should render the focal point field view', () => {
    const { container } = render(<FocalPointView {...props} />);
    expect(container).toMatchSnapshot();
  });
});
