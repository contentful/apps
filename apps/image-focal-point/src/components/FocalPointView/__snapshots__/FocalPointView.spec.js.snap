// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocalPointView should render the focal point field view 1`] = `
<div>
  <div
    class="css-70qvj9"
  >
    <div
      class="TextInput__TextInput___36-K- css-93644d TextInput__TextInput--large___KwY4O TextInput__TextInput--disabled___2t7VS"
    >
      <input
        class="TextInput__TextInput__input___27vDB a11y__focus-border--default___60AXp"
        data-test-id="focal-point"
        disabled=""
        id="focal-point"
        type="text"
        value="x: 10px / y: 30px"
      />
    </div>
    <button
      class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-dp0f2t Button__Button--muted___2Wair"
      data-test-id="cf-ui-button"
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
        tabindex="-1"
      >
        <span
          class="Button__Button__label___3tcOj"
        >
          Set focal point
        </span>
      </span>
    </button>
    <button
      class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 TextLink__TextLink--primary___2Vc9F"
      data-test-id="cf-ui-text-link"
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty"
        tabindex="-1"
      >
        Reset focal point
      </span>
    </button>
  </div>
</div>
`;
